### ISSUE TITLE
Granularity of the guidance: BT, Xpath, other?

### ISSUE THREAD

**Issue by ColinMaudry** [open-contracting/european-union-support#51]
For the standard forms, for each form, we have used the Xpath as the unit of guidance:

- sporadic level to refer to the paper form
- one xpath with its description
- OCDS guidance that may refer to other Xpath.

| Level  |  Xpath | Guidance |
| -----|---------|--------|
| VI.4.2  | Body responsible for mediation procedures <br>`/COMPLEMENTARY_INFO/ADDRESS_MEDIATION_BODY` | Add a party, and add ‘mediationBody’ to its .roles  |

For the eforms guidance, for each form type (Planning, Result, etc.), I'm leaning toward using the BT as the unit of guidance and group the matching Xpath with it:

- BT
- Xpath(s), with description if available
- OCDS guidance that may refer to other BT and Xpath

| BT  |  Xpath | Guidance |
| -----|---------|--------|
| BT-510  | Street name<br>`//cac:PostalAddress/cbc:StreetName` (https://github.com/open-contracting/european-union-support/issues/49#issuecomment-898620088) |  Map to the party's `.address.streetAddress` |

Most BT have a single Xpath, and multi-Xpath BTs seem to have a logic that deserve to be treated as a whole. The information we currently doesn't tell us whether a given BT can have a different logic from a form type or a notice to another.

**Comment by jpmckinney** [comment]
Just to say, all your suggested approaches in this and other issues sound good to me :)

**Comment by ColinMaudry** [comment]
Thanks!

**Comment by ColinMaudry** [comment]
We will start with a guidance centered around BT, perhaps spread to notices belonging to the same form types.

**Comment by ColinMaudry** [comment]
Last week and this week I've been studying alternative approaches to structure the guidance. My initial take was to have it revolve around what I knew best: the annex and its Business Groups and Business Terms.

The problem is that the XML structure is much more complex than the relative flatness of the annex. While the XML of the standard forms was very close to the structure of the actual forms, the XML of eForms relies on a deeper structure with heavy use of cross references. A bit like OCDS when using the lot extension :)

With this in mind, I wonder what would be the best way to present the guidance to the implementers. I fear that guidance that revolves around BT/BG will be quite remote to what developers will deal with: the XML structure of eForms. I have started to extract recurrent "building blocks" from eForms XML, but that'really an "offroad" approach:

- the Publication Office/GROW doesn't provide much documentation on how BT/BG are bound to XML elements
- that complicates the import of the guidance from standard (old forms, \~30%), as the SF/eForms mapping we have relies on BTs, not XML paths

Bottom line: I don't know how to produce guidance that is both convenient for implementers and convenient to produce. I'm thinking of ways to rely on BT/BG BUT highlighting the XML structure.

**Comment by jpmckinney** [comment]
Maybe instead of explicit XML references, we can just have references to one or more XML elements, just to orient the reader, rather than to offer procedural instructions or 1-1 mapping?

**Comment by ColinMaudry** [comment]
Then we stick to the original plan:

| BT/BG | XML paths | guidance |
|---------|-----------|------------|

**Comment by jpmckinney** [comment]
Sounds good to me!
